### DISCUSSION TITLE
Unit field in Open mSupply

### DISCUSSION THREAD

**Discussion by adamdewey** [msupply-foundation/open-msupply#4564] [General]
In mSupply it is mandatory to populate the `unit` field when importing items. 

However the unit field is not exposed in places where you would expect to see it (e.g. in the Stock window, Customer Invoice window etc) 

To get around this most customers include the unit in the _name_ of the item. 

<img width="396" alt="image" src="https://github.com/user-attachments/assets/8afba4ed-0690-462d-91e7-7ba47aee5ad5">

The question, then, is: 

- Does Open mSupply want to
 a) continue the tradition of making users include the unit in the name of the item 
 or 
 b) try and encourage separation of the name and the unit?
 
 If a) then do we need to expose the unit field anywhere in the UI and do we make the unit field non-mandatory?
 If b) which parts of the UI would need to display the unit so that customers felt comfortable enough to not add it to the name?

**Comment by richardmoizeau** [comment]
c) "presentation" & "dosage" (eg. Tablets 500g" should be included in the item name and unit can be entered separately (even if presentation is the same as the unit). Should users don't want to bother with units, they can use a generic "each" (or ea) or else.  

Like Adam, I think it is a UI issue and not a "functional issue". "Unit" should always be displayed where there is a quantity of unit (like the column next to it) so user knows a "quantity of what" every time they look at a table (and also to make the difference with the "quantity of pack") that is often in the same table.

**Reply by adamdewey** [reply]
Perhaps this is getting closer to the mark - is it the presentation that's being included in the name rather than the unit?

**Reply by adamdewey** [reply]
And should there be a separate field for presentation in that case...

**Reply by richardmoizeau** [reply]
No I don't think so, presentation can stay as part the name, to differiente various items (eg Paracetamol) that comes in different presentation (Tab, Inj, etc...) or dosage

**Reply by adamdewey** [reply]
I think that would be covered by item variants?

**Reply by richardmoizeau** [reply]
presentation and dosage are part of the INN (international proprietary name) and item variant main purpose is to differentiate various product under one INN (from different manufacturer, different pack size, etc... or even GTIN code cause each item/manufacturer has a unique one)

**Reply by richardmoizeau** [reply]
> presentation and dosage are part of the INN (international non-proprietary name) and item variant main purpose is to differentiate various product under one INN (from different manufacturer, different pack size, etc... or even GTIN code cause each item/manufacturer has a unique one)

**Reply by adamdewey** [reply]
Oh that's interesting... 

UNICEF / WHO seem to group different presentations under the same item parent for vaccines. 

E.g. 

![image](https://github.com/user-attachments/assets/47f4f25d-ddb9-4605-814b-97854e1cd0d4)

(I was imagining item variants would cater for this scenario too)

**Comment by richardmoizeau** [comment]
By the way, in OG's "stock view", you can customise the layout of the table so that unit appears exactly where you want it to be ;) It also saves the layout so when you open it afterwards, you don't have to customise the table again.

<img width="554" alt="Screenshot 2024-08-13 at 9 38 43 AM" src="https://github.com/user-attachments/assets/0732f106-d33c-4c2d-8ffd-f2e5d82023f3">

**Reply by adamdewey** [reply]
Yes good point - so it's interesting that there is still the tendency to add the unit to the item name. It suggests that the functionality is not fit for purpose for users, so they continue to add it to the name?

**Reply by richardmoizeau** [reply]
cause I think it's still relevant to add the presentation to the item name - but does not mean we should suppress the unit field.

**Comment by mark-prins** [comment]
there is the [item variant](https://docs.google.com/document/d/1fRZslRohRGaZnUo13a2eKZjfO0DGWwXtIFsbEah7A4Y/edit#heading=h.hyld495fuxe5) change which I'm suggesting - that brings in a 'preferred pack size' concept.

I agree with Richard - that we should be displaying unit, and I think that we currently are in oms, and that the separation of name and unit is good ( I _think_ you said this? )

I'm now slightly confused - this is an open mSupply discussion thread, talking about the mSupply UI deficiencies.. are we suggesting changing OG here, or using this as a request to implement in OMS? where my contention is that we are already implementing in OMS

**Reply by adamdewey** [reply]
Ah sorry, I guess I was trying to highlight that even though there is a unit field in OG all users seem to include the unit in the item name because the unit functionality obviously does not do what they need. 

So this ticket is more to try and investigate what that need is, and meet it in Open mSupply, otherwise we'll end up in the same boat as OG.

**Comment by Gwalior-84** [comment]
I struggle to see the value of having Unit as it's own special field.  This is particularly the case when we have other more(?) important information (e.g. strength, and volume) in the Item description.  I contend that:

- The 'Unit' Is not useful for any meaningful report.  I have not come across any management wanting to know:  How many items are delivered as tablets vs capsules or vials instead of ampules.  I would welcome seeing an example to disprove this contention
- The Unit field Is only valuable if the unit detail is not in the Item description.

If the above two contentions are true, then the Unit field unnecessarily occupies screen and report real-estate.
